pkg/testutil/require.go

```diff
@@ -42,3 +43,27 @@ func RequireEqualEmptyNil(t *testing.T, expected, actual interface{}, msgAndArgs
 	msgAndArgs = append(msgAndArgs, cmp.Diff(expected, actual, opts...))
 	require.Truef(t, cmp.Equal(expected, actual, opts...), "Should be equal", msgAndArgs...)
 }
+
+// RequireWithin requires that the runner complete its execution within the specified duration.
+func RequireWithin(t *testing.T, runner func(t *testing.T), timeout time.Duration) {
+	t.Helper()
+
+	ch := make(chan bool, 1)
+	timer := time.NewTimer(timeout)
+	defer timer.Stop()
+
+	go (func() {
+		t.Helper()
+		runner(t)
+		ch <- true
+	})()
+
+	select {
+	case <-timer.C:
+		require.Failf(t, "timed out waiting for runner", "expected to complete in %v", timeout)
+
+	case <-ch:
+		timer.Stop()
```

Comment:
do we to call `timer.Stop()` if you've deferred it above?

Reply:
No, but I'll add it

Reply:
No, but I like being explicit where possible

Reply:
well it could have the effect of being confused